tensorflow/core/kernels/mkl_relu_op.cc

```diff
@@ -1251,6 +1254,76 @@ class MklTanhGradOp : public MklReluGradOpBase<Device, T, eltwise_tanh> {
   }
 };
 
+#define RELU6_UPPER_BOUND 6.0f
+template <typename Device, typename T>
+class MklRelu6Op : public MklReluOpBase<Device, T, eltwise_bounded_relu> {
+ public:
+  ~MklRelu6Op() {}
+
+  explicit MklRelu6Op(OpKernelConstruction* context)
+      : MklReluOpBase<Device, T, eltwise_bounded_relu>(
+            context, RELU6_UPPER_BOUND, 0.0f) {}
+
+  virtual void Compute_Scalar(OpKernelContext* context) {
+    const size_t src_index = 0;  // index of src input tensor
+    const size_t dst_index = 0;  // index of dst output tensor
+    const Tensor& src_tensor = MklGetInput(context, src_index);
+    MklDnnShape dnn_shape_src;
+    GetMklShape(context, src_index, &dnn_shape_src);
+
+    Tensor* dst_tensor = nullptr;
+    void* user_i =
+        static_cast<void*>(const_cast<T*>(src_tensor.flat<T>().data()));
+    MklDnnShape dnn_shape_dst;
+    dnn_shape_dst.SetMklTensor(false);
+    AllocateOutputSetMklShape(context, dst_index, &dst_tensor,
+                              src_tensor.shape(), dnn_shape_dst);
+    void* out_o = static_cast<void*>(dst_tensor->flat<T>().data());
+    (static_cast<T*>(out_o))[0] =
+        std::min(std::max((static_cast<T*>(user_i))[0], static_cast<T>(0)),
+                 static_cast<T>(RELU6_UPPER_BOUND));
+    return;
+  }
+};
+
+template <typename Device, typename T>
+class MklRelu6GradOp
+    : public MklReluGradOpBase<Device, T, eltwise_bounded_relu> {
+ public:
+  ~MklRelu6GradOp() {}
+
+  explicit MklRelu6GradOp(OpKernelConstruction* context)
+      : MklReluGradOpBase<Device, T, eltwise_bounded_relu>(
+            context, RELU6_UPPER_BOUND, 0.0f) {}
+
+  virtual void Compute_Scalar(OpKernelContext* context) {
+    const size_t diff_dst_index = 0;  // index of diff_dst input tensor
+    const size_t src_index = 1;       // index of src input tensor
+    const size_t diff_src_index = 0;  // index of diff_src output tensor
+    const Tensor& src_tensor = MklGetInput(context, src_index);
+    const Tensor& diff_dst_tensor = MklGetInput(context, diff_dst_index);
+    Tensor* diff_src_tensor = nullptr;
+
+    MklDnnShape dnn_shape_diff_dst;
+    GetMklShape(context, diff_dst_index, &dnn_shape_diff_dst);
+
+    MklDnnShape dnn_shape_diff_src;
+    dnn_shape_diff_src.SetMklTensor(false);
+    AllocateOutputSetMklShape(context, diff_src_index, &diff_src_tensor,
+                              diff_dst_tensor.shape(), dnn_shape_diff_src);
+    void* out_o = static_cast<void*>(diff_src_tensor->flat<T>().data());
```

Comment:
Same question about pointers, so much syntactic noise but I don't get why it's needed here,

Reply:
Done by code change. thanks.